docs/tutorials/developer-journey/level-1/1.3-first-dapp.md

```diff
@@ -0,0 +1,723 @@
+  //This is so the default behavior is set BEFORE the awaits are called below
+  e.preventDefault();
+ 
+  // Query the question from the backend
+  const question = await poll_backend.getQuestion();
+  document.getElementById("question").innerText = question;
+
+  //Query the vote counts for each option
+  // Example JSON that the frontend will get using the values above
+  // [["Motoko","0"],["Python","0"],["Rust","0"],["TypeScript","0"]]
+  const voteCounts = await poll_backend.getVotes();
+  updateLocalVoteCounts(voteCounts);
+  displayResults();
+  return false;
+}, false);
+
+//Event listener that listens for when the form is submitted.
+//When the form is submitted with an option, it calls the backend canister
+//via "await poll_backend.vote(selectedOption)"
+pollForm.addEventListener('submit', async (e) => {
+  //note that this is at beginning of the submit callback, this is deliberate
+  //This is so the default behavior is set BEFORE the awaits are called below
+  e.preventDefault(); 
+
+  const formData = new FormData(pollForm);
+  const checkedValue = formData.get("option");
+
+  const updatedVoteCounts = await poll_backend.vote(checkedValue);
+  console.log("Returning from await...")
+  console.log(updatedVoteCounts);
+  updateLocalVoteCounts(updatedVoteCounts);
+  displayResults();
+  return false;
+}, false);
+
+resetButton.addEventListener('click', async (e) => {
+
+    e.preventDefault();
+    
+    //Reset the options in the backend
+    await poll_backend.resetVotes();
+    const voteCounts = await poll_backend.getVotes();
+    updateLocalVoteCounts(voteCounts);
+
+    //re-render the results once the votes are reset in the backend
+    displayResults();
+    return false;
+}, false);
+
+//3. HELPER FUNCTIONS
+
+//Helper vanilla JS function to create the HTML to render the results of the poll
+function displayResults() {
+  let resultHTML = '<ul>';
+  for (let key in pollResults) {
+      resultHTML += '<li><strong>' + key + '</strong>: ' + pollResults[key] + '</li>';
+  }
+  resultHTML += '</ul>';
+  resultsDiv.innerHTML = resultHTML;
+};
+
+//This helper updates the local JS object that teh browser holds
+// Example JSON that the frontend will get using the values above
+  // [["Motoko","0"],["Python","0"],["Rust","0"],["TypeScript","0"]]
+function updateLocalVoteCounts(arrayOfVoteArrays){
+
+  for (let voteArray of arrayOfVoteArrays) {
+    //Example voteArray -> ["Motoko","0"]
+    let voteOption = voteArray[0];
+    let voteCount = voteArray[1];
+    pollResults[voteOption] = voteCount;
+  }
+
+};
+```
+
+:::info
+#### What does this code do?
+
+On line 4, the statement `import { poll_backend } from "../../declarations/poll_backend";` is what allows the frontend to import an interface for the backend canister and seamlessly send it via messages using Candid. This line is directly related to the following lines where the frontend JS talks to the backend:
```

Comment:
```suggestion
On line 4, the statement `import { poll_backend } from "../../declarations/poll_backend";` is what allows the frontend to import an interface for the backend canister and seamlessly interact with it. This line is directly related to the following lines where the frontend JS talks to the backend:
```

Reply:
In general, I think this is a bit overwhelming for newcomers (both web2 and web3).

It might be better to start with the core js snippet on how to call a canister method from JS and explain it and then provide the complete code afterward.